### ISSUE TITLE
Parsing error with constructor property promotion and @var docblock

### ISSUE THREAD

**Issue by archfz** [phpmd/phpmd#896]
- PHPMD version: 2.10.1
- PHP Version: 8.0.5
- Installation type: composer
- Operating System / Distribution & Version: Ubuntu 20.04

### Current Behavior
PHPMD fails to parse file where constructor property promotion is used in combination with @var docblock.

```php
class Test {
   public function __construct(
      private string $id,
      /** @var array<array<string>|null> */
      private array $complex
   ) {}
}
```

Error

```
ERROR while parsing 
--------------------
Unexpected token: private, line: 13
```

### Expected Behavior
It should not error.

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by adrienlucas** [comment]
This is fixed but not yet released, see https://github.com/phpmd/phpmd/issues/889
(TL;DR; `composer require --dev "pdepend/pdepend:2.x-dev"` to solve the parse error; but you may still get a `unused parameter` false-positive)

**Comment by kylekatarnls** [comment]
You can update to last pdepend/pdepend now to get it fixed, phar version of 2.10.2 that will have it will be released this week too.
